comp/core/workloadmeta/collectors/util/ecs_util.go

```diff
@@ -0,0 +1,267 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+
+package util
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/v3or4"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// TaskParser is a function that parses ECS tasks
+type TaskParser func(ctx context.Context) ([]workloadmeta.CollectorEvent, error)
+
+// IsTaskCollectionEnabled returns true if the task metadata collection is enabled for core agent
+// If agent launch type is EC2, collector will query the latest ECS metadata endpoint for each task returned by v1/tasks
+// If agent launch type is Fargate, collector will query the latest ECS metadata endpoint
+func IsTaskCollectionEnabled() bool {
+	return config.Datadog.GetBool("ecs_task_collection_enabled") && (flavor.GetFlavor() == flavor.DefaultAgent)
+}
+
+// ParseV4Task parses a metadata v4 task into a workloadmeta.ECSTask
+func ParseV4Task(task v3or4.Task, seen map[workloadmeta.EntityID]struct{}) []workloadmeta.CollectorEvent {
```

Comment:
Why are these functions (ParseV4Task, and ParseV4TaskContainers) extracted from the `comp/core/workloadmeta/collectors/internal/ecs/v4parser.go` package? The util package usually is dedicated for functions that are shared by multiple packages? These functions are only used in `comp/core/workloadmeta/collectors/internal/ecs/v4parser.go`. I would suggest moving them to that package

Reply:
https://github.com/DataDog/datadog-agent/pull/21836#discussion_r1567042405